tests/endtoend/.dockerignore

```diff
@@ -1,7 +0,0 @@
-# Ignore everything
```

Comment:
Sorry if I'm missing something obvious, but why do we need to build the E2E test Docker image in the root of the repository now? From reading the PR I don't see where we're adding code from outside the E2E test directory.

Reply:
I initially tried to build the test runner in the context of the e2e directory, but this sort of runs into issues with how Go expects projects to be structured. For example, there needs to a `go.mod` file somewhere, and while it is possible to have multiple `go.mod` files,  from what I've read it's not generally recommended and is simpler to have 1 `go.mod` file per repository.

As for why we can't just copy over the `go.mod` file for the test image, Docker enforces a security boundary on the root of the image's context and doesn't allow you to copy files from parent directories. So it ended up being simpler to just take a snapshot of the entire repository with respect to the root and run tests from there.

Reply:
Sounds good to me